store/types/utils.go

```diff
@@ -105,3 +108,68 @@ func PrefixEndBytes(prefix []byte) []byte {
 func InclusiveEndBytes(inclusiveBytes []byte) []byte {
 	return append(inclusiveBytes, byte(0x00))
 }
+
+// Marshaler defines a interface needed for the store package to marshal data
+type Marshaler interface {
```

Comment:
```suggestion
type Codec interface {
```
Why not just call `Codec`? It's the expected sub-interface of `Codec` we depend on.

Reply:
changed and added unmarshal for completeness with marshal.